File: src/main/java/ru/develgame/jsfgame/common/LoggerProducer.java
package ru.develgame.jsfgame.common;

import jakarta.enterprise.inject.Produces;
import jakarta.enterprise.inject.spi.InjectionPoint;
import jakarta.inject.Named;
import jakarta.inject.Singleton;
import java.util.logging.Logger;

@Named
@Singleton
public class LoggerProducer {
    @Produces
    public Logger produceLogger(InjectionPoint injectionPoint) {
        return Logger.getLogger(injectionPoint.getBean().getBeanClass().getName());
    }
}

File: src/main/kotlin/ru/develgame/jsfgame/LoggerProducer.kt
package ru.develgame.jsfgame

import java.util.logging.Logger
import javax.enterprise.inject.Produces
import javax.enterprise.inject.spi.InjectionPoint
import javax.inject.Named
import javax.inject.Singleton

@Named
@Singleton
open class LoggerProducer {
    @Produces
    open fun produceLogger(injectionPoint: InjectionPoint): Logger {
        return Logger.getLogger(javaClass.name)
    }
}